rs/xnet/payload_builder/src/lib.rs

```diff
@@ -712,6 +712,22 @@ impl XNetPayloadBuilderImpl {
                     ));
                 }
             }
+
+            // Ensure the signal limit is respected.
+            let signal_limit = get_signal_limit(Some(expected), slice.header().begin());
+            if messages.len() > signal_limit {
```

Comment:
The logic why this can be done separately from the check above about `msg_limit` is that we will include the `min` of `signal_limit` and `msg_limit` so for an honestly built block we know that both must pass right?